test/DistributionTest.js

```diff
@@ -0,0 +1,151 @@
+            expect(await instanceDistribution.balanceOfBeneficiares(acc4)).to.be.bignumber.equal(ether('30'));
+        });
+
+    });
+
+    describe("Decrease reward function", async () => {
+
+        it("Decrease reward - must reverted , amount = 0", async () => {
+            await expectRevert(instanceDistribution.decreaseReward(acc3, new BN(0)), "beneficiary or amount equal to 0");
+        });
+
+        it("Decrease reward - must reverted , address = 0", async () => {
+            let addr = constants.ZERO_ADDRESS;
+            await expectRevert(instanceDistribution.decreaseReward(addr, ether('30')), "beneficiary or amount equal to 0");
+        });
+
+        it("Decrease reward - must reverted , caller is not the owner ", async () => {
+            await expectRevert(instanceDistribution.decreaseReward(acc3, ether('30'), { from: acc2 }), "caller is not the owner");
+        });
+
+        it("Decrease reward - must success", async () => {
+            await instanceDistribution.decreaseReward(acc3, ether('30'));
+            expect(await instanceDistribution.balanceOfBeneficiares(acc3)).to.be.bignumber.equal(ether('80'));
+        });
+    });
+
+    describe("withdraw function", async () => {
+
+        it("Emergency withdraw - call from not owner reverted", async () => {
+            await expectRevert(instanceDistribution.emergencyWithdraw(ether('200'), { from: acc2 }), "caller is not the owner");
+        });
+
+        it("Emergency withdraw", async () => {
+            await instanceDistribution.emergencyWithdraw(ether('200'));
+            expect(await instanceToken.balanceOf(owner)).to.be.bignumber.equal(ether('800'));
+        });
+
+    });
+
+    describe("claim", async () => {
+
+        it("Lock/unlock + claim - must reverted , claim is locked ", async () => {
+            await instanceDistribution.lockRewards(false);
+            expect(await instanceDistribution.lockStatus()).to.be.eq(false);
+            await expectRevert(instanceDistribution.claim({ from: acc2 }), 'claim is locked');
+        });
+
+        it("Lock/unlock + claim - balance 0", async () => {
+            await instanceDistribution.lockRewards(true);
+            expect(await instanceDistribution.lockStatus()).to.be.eq(true);
+            await expectRevert(instanceDistribution.claim({ from: owner }), "balance must be > 0");
+        });
+
+        it("Lock/unlock + claim - success", async () => {
+            let lock = await instanceDistribution.lockRewards(true);
+            expectEvent(lock, "LockStatus", { lock: true });
+            expect(await instanceDistribution.lockStatus()).to.be.eq(true);
+            let claim = await instanceDistribution.claim({ from: acc2 });
+            expectEvent(claim, "ClaimInfo", { beneficiaries_: acc2, amount_: ether('10') });
+            expect(await instanceToken.balanceOf(acc2)).to.be.bignumber.equal(ether('10'));
```

Comment:
also check balance of acc2 in start of test.
also check balanceOfBeneficiares for acc2 in end of tests